From 25150436a761d32a4bec1acf9834802215dc6f90 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 22 Jul 2016 22:58:57 +0000
Subject: [PATCH 4/5] ARM: dts: k2l: add dts node for k2l hw_rng driver

This patch adds dts node for k2l hw_random generator driver

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2l-netcp.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
index 04ba9f3..14f544b 100644
--- a/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2l-netcp.dtsi
@@ -511,3 +511,23 @@ netcp: netcp@26000000 {
 		};
 	};
 };
+
+sa_subsys: subsys@26080000 {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	ranges = <0 0x26080000 0x40000>;
+	compatible = "simple-bus";
+
+	sa_config: subsys {
+		compatible      = "syscon";
+		reg             = <0x0 0x100>;
+	};
+
+	hwrng {
+		compatible = "ti,keystone-rng";
+		sa-cfg = <&sa_config>;
+		clocks = <&clksa>;
+		clock-names = "fck";
+		reg = <0x24000 0x1000>;
+	};
+};
-- 
1.9.1

